2.train-VAE/betavae.py

```diff
@@ -0,0 +1,115 @@
+#Code based on https://github.com/1Konny/Beta-VAE/blob/master/model.py
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+
+class BetaVAE(nn.Module):
+    """
+    Beta Variational Autoencoder (VAE) class.
+
+    Args:
+        input_dim (int): Dimension of the input data.
+        hidden_dim (int): Dimension of the hidden layer. (this is utilized if two layers)
+        latent_dim (int): Dimension of the latent space.
+        beta (float): Weight for the Kullback-Leibler divergence term in the loss function.
+    """
+    def __init__(
+        self, 
+        input_dim, 
+        #hidden_dim,
+        latent_dim, 
+        beta):
+        super(BetaVAE, self).__init__()
+        self.encoder = nn.Linear(input_dim, latent_dim * 2)
+        self.decoder = nn.Sequential(
+            nn.Linear(latent_dim, input_dim),
+            nn.Sigmoid()
+        )
+        self.latent_dim = latent_dim
+        self.beta = beta
+
+    def reparameterize(self, mu, log_var):
+        """
+        Reparameterize from N(mu, var) to N(0,1)
+        """
+        std = torch.exp(0.5 * log_var)
+        eps = torch.randn_like(std)
+        return mu + eps * std
+
+    def forward(self, x):
+        """
+        Forward pass through the VAE.
+
+        Args:
+           
```

Comment:
Make sure docstrings are consistently formatted. We have to show you [precommit](https://pre-commit.com/) and autoformating! ([black](https://github.com/psf/black) is an example, but I'm not sure what the lab uses now)